lab_2_retrieval_w_bm25/main.py

```diff
@@ -77,6 +122,23 @@ def calculate_idf(vocab: list[str], documents: list[list[str]]) -> dict[str, flo
 
     In case of corrupt input arguments, None is returned.
     """
+    if (not isinstance(vocab, list) or not all(isinstance(token, str) for token in vocab)
+            or not vocab):
+        return None
+    if (not isinstance(documents, list) or not all(isinstance(doc, list) for doc in documents)
+            or not documents):
+        return None
+    if not all(isinstance(word, str) for doc in documents for word in doc):
+        return None
+    idf = {}
+    for word in vocab:
+        num = 0
+        for doc in documents:
+            if word in doc:
+                num += 1
```

Comment:
how else can we count the number of elements included in doc?

Reply:
I have no idea((